### PR TITLE
Add dbt tests for the reporting views/CTAS

### PR DESCRIPTION
**Pull request by wrridgeway**
This PR is primarily about adding tests for the reporting DB to dbt, but there are also a few minor changes to a reporting CTAS and view that are a direct result of implementing these new tests.

### CODE REVIEW

File: dbt/models/reporting/schema.yml [hunk 1]
```diff
@@ -1,32 +1,157 @@
 sources:
   - name: reporting
-    tags:
-      - load_auto
     tables:
       - name: ratio_stats
-        description: '{{ doc("table_ratio_stats") }}'
```
Comment (dfsnow) [inline dbt/models/reporting/schema.yml:7]: **issue:** The table-level descriptions shouldn't be removed.

File: dbt/models/reporting/schema.yml [hunk 2]
```diff
@@ -1,32 +1,157 @@
...
+                  AND within_10_pct >= within_05_pct
 
 models:
-  - name: reporting.res_report_summary
-    description: '{{ doc("table_res_report_summary") }}'
-    config:
-      tags:
-        - daily
-
+  - name: res_report_summary
```
Comment (dfsnow) [inline dbt/models/reporting/schema.yml:34]: **issue (blocking):** The name key here needs to be prefixed with the schema. Also, don't remove the tags and description.

Note the warnings in this workflow under "Build models": https://github.com/ccao-data/data-architecture/actions/runs/6869938167/job/18683780160?pr=218

File: dbt/models/reporting/schema.yml [hunk 3]
```diff
@@ -1,32 +1,157 @@
...
+      - unique_combination_of_columns:
+          name: vw_pin_most_recent_sale_unique_by_pin
+          combination_of_columns:
+            - pin
+      - value_is_present:
+          name: vw_pin_most_recent_sale_null
+          expression: sale_price IS NULL
+      - value_is_present:
+          name: vw_pin_most_recent_sale_not_null
+          expression: sale_price IS NOT NULL
```
Comment (dfsnow) [inline dbt/models/reporting/schema.yml:95]: **issue:** Can you add some context about what this is testing in a YAML comment?

File: dbt/models/reporting/schema.yml [hunk 4]
```diff
@@ -1,32 +1,157 @@
...
+        expression: >-
+          triad IS NOT NULL
+            AND geography_type IS NOT NULL
+            AND property_group IS NOT NULL
+            AND assessment_stage IS NOT NULL
+        select_columns:
+          - triad
+          - geography_type
+          - property_group
+          - assessment_stage
```
Comment (dfsnow) [inline dbt/models/reporting/schema.yml:60]: **nitpick:** Let's add a line break between models for clarity of formatting:

```suggestion
          - assessment_stage
          
```

**Review by dfsnow — CHANGES_REQUESTED** [review]

File: dbt/models/reporting/schema.yml [hunk 5]
```diff
@@ -1,32 +1,157 @@
 sources:
   - name: reporting
-    tags:
-      - load_auto
```
[the remark this answers is not in the document]
Comment (wrridgeway) [reply dbt/models/reporting/schema.yml:4]: This is a mistake

File: dbt/models/reporting/schema.yml (hunk 1 above)
Comment (wrridgeway) [reply dbt/models/reporting/schema.yml:7]: Mistake as well

File: aws-athena/views/reporting-vw_assessment_roll.sql
[the remark this answers is not in the document]
Comment (wrridgeway) [reply aws-athena/views/reporting-vw_assessment_roll.sql:1]: Adding the `NOT NULL` conditionals did remove some rows from these views and tables. There were previously some rows affiliated with `NULL` values for aggregated categories like township and triad that are a product of bad data.

Changing from `UNION` to `UNION ALL` is just to make sure the views are doing what they were originally intended to do.

File: aws-athena/ctas/reporting-res_report_summary.sql
[the remark this answers is not in the document]
Comment (wrridgeway) [reply aws-athena/ctas/reporting-res_report_summary.sql:1]: See below

**Comment by wrridgeway** [comment]
@dfsnow should be ready for review again

**Review by dfsnow — APPROVED** [review]
Alright @wrridgeway, this now looks good to go!

**Pull request merged**